File: javascript/grade-school/school.js
module.exports = function School() {
  return {
    db: {},

    add: function(name, grade) {
      var current = this.grade(grade);
      current.push(name);
      this.db[grade] = current;
    },

    grade: function(grade) {
      return this.db[grade] || [];
    },

    sort: function() {
      return Object.keys(this.db).reduce(function(db, grade) {
        db[grade] = db[grade].sort();
        return db;
      }, this.db);
    }
  };
};

File: clojure/grade-school/school.clj
(ns school)

(defn grade [db level]
  (get db level []))

(defn add [db name level]
  (let [students (grade db level)]
    (assoc db level (conj students name))))

(defn sorted [db]
  (into {} (map (fn [[k v]] [k (sort v)]) db)))
